### ISSUE TITLE
Int32 offset overflow in the compute-delta reduction silently corrupts dQ/dK for long sequences

### ISSUE THREAD

**Issue by luowyang** [SHI-Labs/NATTEN#347]
For long sequences, `na1d` backward returns finite but wrong `dq`/`dk`. No error, no warning.

## Repro

```python
import torch, natten

B, H, D, KERNEL = 1, 16, 64, 511

def grads(Q, pt_reduction):
    g = torch.Generator(device="cuda").manual_seed(0)
    shape, kw = (B, Q, H, D), dict(generator=g, device="cuda", dtype=torch.float16)
    q = torch.randn(shape, **kw, requires_grad=True)
    k = torch.randn(shape, **kw, requires_grad=True)
    v = torch.randn(shape, **kw, requires_grad=True)
    grad_out = torch.randn(shape, **kw)
    out = natten.na1d(q, k, v, kernel_size=KERNEL, backend="cutlass-fna",
                      backward_use_pt_reduction=pt_reduction)
    out.backward(grad_out)
    return q.grad.float(), k.grad.float()

for Q in (2_097_152, 2_097_153):     # last safe length, then one token more
    dq, dk = grads(Q, False)         # CUTLASS compute-delta kernel (the default)
    rq, rk = grads(Q, True)          # same delta via the PyTorch reduction
    print(Q, (dq - rq).abs().max().item(), (dk - rk).abs().max().item())
```

```text
2097152 0.0009765625       0.0009765625
2097153 0.0542144775390625 0.0079345703125
```

`0.0009765625` is the noise floor: running the PyTorch reduction twice against identical inputs gives the same number. One extra token pushes `dq` 55x past it.

## Cause

`FmhaKernelBwdSumOdO::Arguments` stores the Q-stride components of `O`, `dO` and `delta` as `int`, so the device-side offset

```cpp
ptr_O_bh + idx_q * get<2>(params.stride_O)
```

is computed in 32-bit. For heads-last `[B, Q, H, D]` the `O`/`dO` Q stride is `H * D`. The stride itself is small, but the *offset* overflows once

```text
(Q - 1) * H * D > INT32_MAX
```

At `H=16, D=64` the last safe length is exactly `Q = 2_097_152` (largest element index `2_097_151 * 1024 + 1023 == INT32_MAX`), and `Q = 2_097_153` is the first that overflows. Nothing checks the size on the way in.

## Notes

- Only on the `backward_use_pt_reduction=False` path, which is the API default.
- `dv` is bit-identical at every length — `delta` only feeds the dQ/dK softmax correction.
- It gets worse with length: at `Q=2_300_000` the deviations are `0.829` / `0.880`.
- Silent in the range we tested; an invalid pointer offset is possible in principle.
- #337 widened the CUTLASS FNA iterator strides for the same class of overflow. This standalone reduction kernel wasn't covered by it.
- Seen on SM90, CUDA 12.8, PyTorch 2.11, NATTEN 0.21.7, fp16.
- Crossing the boundary needs just over 4 GiB for each of `O` and `dO`, so the repro wants a large-memory GPU and is not CI-sized.
